compiler/rustc_const_eval/messages.ftl

```diff
@@ -249,18 +232,47 @@ const_eval_mutable_ref_escaping =
         If you really want global mutable state, try using an interior mutable `static` or a `static mut`.
 
 const_eval_nested_static_in_thread_local = #[thread_local] does not support implicit nested statics, please create explicit static items and refer to them instead
+
+const_eval_non_const_await =
+    cannot convert `{$ty}` into a future in {const_eval_const_context}s
+
+const_eval_non_const_closure =
+    cannot call {$non_or_conditionally}-const closure in {const_eval_const_context}s
```

Comment:
this is quite... annoying

ideally fluent messages don't use any text segments given that they exist for translateability. Given the current state of our translation effort, this is non-blocking

Reply:
Well I kinda don't want to migrate this back to non translatable infra, so this will be in a hacky middle state for now.